workload/src/main/java/com/google/finapp/WorkloadClient.java

```diff
@@ -14,28 +14,91 @@
 
 package com.google.finapp;
 
+import com.google.common.collect.ImmutableList;
+import com.google.finapp.CreateAccountRequest.Status;
 import com.google.finapp.FinAppGrpc.FinAppBlockingStub;
 import com.google.protobuf.ByteString;
 import io.grpc.ManagedChannel;
 import io.grpc.StatusRuntimeException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-/** A gRPC client for the finance sample app. */
-public class WorkloadClient {
+/** A gRPC client for the finance sample app. Performs a series of tasks (gRPC methods) in order. */
+public class WorkloadClient implements Runnable {
 
   private final FinAppBlockingStub blockingStub;
```

Comment:
Another interesting thing to consider: since we're reusing the channel across all threads and using blocking stubs the bottleneck here might be gRPC. I wonder if you get higher QPS with non-blocking stubs or separate channels per worker.

Reply:
#55